src/coreclr/jit/hwintrinsiccodegenarm64.cpp

```diff
@@ -559,7 +567,117 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
                     }
 
                     break;
+                }
+                case 3:
+                {
+                    assert(instrIsRMW);
+                    assert(HWIntrinsicInfo::IsFmaIntrinsic(intrinEmbMask.id));
+                    assert(falseReg != embMaskOp1Reg);
+                    assert(falseReg != embMaskOp2Reg);
+                    assert(falseReg != embMaskOp3Reg);
+
+                    bool useAddend = true;
+                    if (targetReg == embMaskOp2Reg)
+                    {
+                        useAddend = false;
+                        std::swap(embMaskOp1Reg, embMaskOp2Reg);
+                    }
+                    else if (targetReg == embMaskOp3Reg)
+                    {
+                        useAddend = false;
+                        std::swap(embMaskOp1Reg, embMaskOp3Reg);
+                    }
```

Comment:
nit: These seem like they could do with a comment as to why its safe/correct to swap. -- I'm not sure they actually are correct as is.

In particular the intrinsic is:
* `op1 == addend`
* `op2 == multiplicand`
* `op3 == multiplier`

Then we have:
* FMLA is `reg1 = reg1 + (reg2 * reg3)`
* FMAD is `reg1 = (reg1 * reg2) + reg3`

Given that:
* when `targetReg == op1Reg`, then things are already fine and need no sorting, we will emit `FMLA`
* when `targetReg == op2Reg`, then we need to swap op2 and op3, then op1 and op3, allowing us to emit `FMAD`
  * if we just swap `op1` and `op2` like you're doing, the addend ends in `reg2` and the result will be wrong
* when `targetReg == op3Reg`, then we need to swap op1 and op3, allowing us to emit `FMAD`

So I think this rather needs to be:
```suggestion
                     if (targetReg == embMaskOp2Reg)
                    {
                        useAddend = false;
                        std::swap(embMaskOp2Reg, embMaskOp3Reg);
                        std::swap(embMaskOp1Reg, embMaskOp3Reg);
                    }
                    else if (targetReg == embMaskOp3Reg)
                    {
                        useAddend = false;
                        std::swap(embMaskOp1Reg, embMaskOp3Reg);
                    }
```

Reply:
ah yes, I missed a case for `targetReg == embMaskOp2Reg`. Will fix it along with the comments.

Reply:
For `targetReg == op2Reg` case, alternatively we can also have:
1. `reg2 = (reg2 * reg3) + reg1`
2. `reg2 = (reg2 * reg1) + reg3`  ; swap `reg1 <--> reg3`
3. `reg1 = (reg1 * reg2) + reg3` ; swap `reg1 <--> reg2`
4. Use `FMAD`

Not that one is better than other, but just want to make sure that I am getting the calculations right.